lib/rocket_chat/server.rb

```diff
@@ -45,5 +45,9 @@ def login(username, password)
       )
       Session.new self, Token.new(response['data'])
     end
+
+    def request_json(path, options = {})
+      super(path, @options.merge(options))
```

Comment:
You're right. Pity there is no deep_merge as part of Ruby. There are enough gems and examples around, though.